ui/hooks/gasFeeInput/useGasFeeInputs.js

```diff
@@ -165,9 +169,24 @@ export function useGasFeeInputs(
   useEffect(() => {
     if (supportsEIP1559) {
       if (transaction?.userFeeLevel) {
-        setEstimateUsed(transaction?.userFeeLevel);
         setInternalEstimateToUse(transaction?.userFeeLevel);
       }
+
+      const maximumGas = new Numeric(transaction?.txParams?.gas ?? '0x0', 16)
+        .times(new Numeric(transaction?.txParams?.maxFeePerGas ?? '0x0', 16))
+        .toPrefixedHexString();
+
+      const fee = new Numeric(maximumGas, 16, EtherDenomination.WEI)
+        .toDenomination(EtherDenomination.ETH)
+        .toBase(10)
+        .toString();
+
+      if (Number(fee) > Number(GAS_LIMIT_TOO_HIGH_IN_ETH)) {
+        setEstimateUsed(PriorityLevels.dappSuggestedHigh);
+      } else if (transaction?.userFeeLevel) {
+        setEstimateUsed(transaction?.userFeeLevel);
+      }
+
```

Comment:
Will be nice to test cover above code also.

Reply:
I have another issue related to this PR: https://github.com/MetaMask/metamask-extension/issues/19305

I will like to tackle these comments on that issue and close this PR (it's taking too long and I don't think these are blocking requests)